src/components/Terminal/ProblemsTab.tsx

```diff
@@ -1,73 +1,299 @@
+    <div className='flex flex-row items-center'>
+      <span className="text-xs">Diagnostics</span>
+      <ul className='flex flex-row items-center'>
+        {errorDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1.5 text-xs bg-red-500'>
+              {errorDiagnostics.length}
+            </span>
+          </li>
+        )}
+        {warningDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1 text-xs bg-yellow-500'>
+              {warningDiagnostics.length}
+            </span>
+          </li>
+        )}
+        {infoDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1 text-xs bg-blue-500'>
+              {infoDiagnostics.length}
+            </span>
+          </li>
+        )}
+        {hintDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1 text-xs bg-blue-500'>
+              {hintDiagnostics.length}
+            </span>
+          </li>
+        )}
+      </ul>
+    </div>
+  );
+};
+
+interface SeverityIconProps {
+  severity: Diagnostic['severity']
+}
+
+const SeverityIcon: React.FunctionComponent<SeverityIconProps> = ({ severity }) => {
+  switch (severity) {
+  case 1: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscWarning className='text-yellow-500 w-4 h-4' />
     </div>
   );
+  case 2: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscInfo className='text-blue-500 w-4 h-4' />
+    </div>
+  );
+  case 3: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscLightbulb className='text-green-500 w-4 h-4' />
+    </div>
+  );
+  default: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscError className='text-red-500 w-4 h-4' />
+    </div>
+  );
+  }
+};
+
+type ActiveSeverity = DiagnosticSeverity | 'all';
+
+interface SeverityButtonsProps {
+  diagnostics: Diagnostic[];
+  active: ActiveSeverity;
+  setActive: React.Dispatch<React.SetStateAction<ActiveSeverity>>;
+}
+
+const SeverityButtons: React.FunctionComponent<SeverityButtonsProps> = ({ diagnostics, active, setActive: defaultSetActive }) => {
+  console.log(diagnostics);
+
+  const errorDiagnostics = diagnostics.filter(diagnostic => diagnostic.severity === 0);
+  const warningDiagnostics = diagnostics.filter(diagnostic => diagnostic.severity === 1);
+  const infoDiagnostics = diagnostics.filter(diagnostic => diagnostic.severity === 2);
+  const hintDiagnostics = diagnostics.filter(diagnostic => diagnostic.severity === 3);
+
+  const activeBg = 'bg-gray-900';
+  const notActiveBg = 'bg-gray-700';
+
+  const setActive = useCallback((type: ActiveSeverity) => {
+    defaultSetActive(type === active ? 'all' : type);
+  }, [active]);
+
+  return (
+    <ul className='flex flex-row items-center'>
```

Comment:
Why are you using `<ul>` if then you're not using `<li>` inside? 🤔

Reply:
I don't know, probably I missed, fixed 😅